Getting-Started-with-Akka-http-signature.md

```diff
@@ -0,0 +1,232 @@
+  public exponent: 65537
+priv: java.security.interfaces.RSAPrivateKey = sun.security.rsa.RSAPrivateCrtKeyImpl@ffd26868
+```
+
+This command will generate a Public and Private key pair. As we can see only information about the Public key is being displayed in BigInt notation. The private key will be hidden from view - only the reference memory location will be shown.
+
+From here many different representations can be shown of the modulus of the key.
+One can show a String representation of the key modulus by using:
+```scala
+@ val stringPublic = pub.getModulus.toString
+
+stringPublic: String = """Sun RSA public key, 2048 bits
+  modulus: 18617616591683306896114159596818930151392321021655743092554571476906272216221226120621265951005438129779697999635487320523329319543618461573656972895570742121788481588357965539288547158041270238222755734178260848243117538931191184092136701393910768801017892322044665470799614169981926296276068464740129396666480935344897491410772746293264767223181696074228079861695236831569819678997680346803798366156295040872828579274684209593397084757356046266872087618604860475724560124477904326733099073845912710038077198861072318221134249054814455997792289530305256172821638522445655947916737229103835394051258962680413560258903
+  public exponent: 65537"""
+
+``` 
+
+The keys' moduli can also be displayed in hexadecimal format:
+
+```scala
+@ val hexaPublic = pub.getModulus.toString(16)
+
+hexaModul : String = "937adccd722bc982aed4847872b81e36b890bca13166714bc2befe4d8547b6218ecd2da1eb020198a4ea00e4db6757c7dda738ec8db8b3bf211d3a3a17e196a2035bc4c79d06d8a581487d9f49e86374712b10ef500dfa242a20cab52911e2636c9d99b21fe9768ef2381989a25dc8b0b7a46531249aac27c4b8ab451a19d5fbdfa5f78b0deac9778c7ff87cf6106ae4a6433466beb21df1265bf1fc9ab9cc80d7aff8cc4d0f67ae28647e4048da8df753493b8de6a8e0961416b4b37f7012907d2e756034b8a84c6e495c8f1d81f69843ae51379571d83b38e1c08a08c3748ef75ed7ecb016d0c8426b30c8c5060a08f87f6764b0ec14667cd6f0daa1244157"
+```
+The RSAKeys class' function .save() will return a String representation of the key in Base64 format
+
+```scala
+@ val basePublic = RSAKeys.save(pub)
+
+basePublic: String = """MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAk3rczXIryYKu1IR4crgeNriQvKExZnFL
+wr7+TYVHtiGOzS2h6wIBmKTqAOTbZ1fH3ac47I24s78hHTo6F+GWogNbxMedBtilgUh9n0noY3Rx
+KxDvUA36JCogyrUpEeJjbJ2Zsh/pdo7yOBmJol3IsLekZTEkmqwnxLirRRoZ1fvfpfeLDerJd4x/
++Hz2EGrkpkM0Zr6yHfEmW/H8mrnMgNev+MxND2euKGR+QEjajfdTSTuN5qjglhQWtLN/cBKQfS51
+YDS4qExuSVyPHYH2mEOuUTeVcdg7OOHAigjDdI73XtfssBbQyEJrMMjFBgoI+H9nZLDsFGZ81vDa
+oSRBVwIDAQAB"""
+```
+
+Keys can also be given different references: 
```

Comment:
the point of this exercise was not to make a new reference to the key, but to show that one can convert the key to a PEM formatted string and then convert it back to the same `==` object by reading that string. Ie. one can use the same public key another day.